src/typegoose.ts

```diff
@@ -251,9 +245,6 @@ export function getDiscriminatorModelForClass<T, U extends AnyParamConstructor<T
   id?: string
 ) {
   const name = getName(cl);
-  if (models.get(name)) {
```

Comment:
why not return the cached one?

Reply:
Because classes can be used in different ways. 
Eg. as a top-level model to save data directly into a collection and as a nested class or array element class. The resulting models are different depending on the parent model.

Reply:
but then when it is called again, the model in "models" will be overwritten?